libs/external-router/src/token/type-resolvable-routes.token.ts

```diff
@@ -0,0 +1,30 @@
+import { InjectionToken } from '@angular/core';
+import { TypeRoutePair } from '../model/type-route-pair';
+import { RouteWithoutPath } from '../model/route-without-path';
+
+import { Provider } from '@angular/core';
+import { DaffExternalRouterType } from '../model/route-type';
+
+/**
+ * A multi-token that allows you to register route "types" that correspond to routes.
+ */
+export const DAFF_EXTERNAL_ROUTER_TYPE_RESOLVABLE_ROUTES = new InjectionToken<
+	TypeRoutePair[]
+>('DAFF_EXTERNAL_ROUTER_RESOLVABLE_TYPE_ROUTES', {
+	providedIn: 'root',
```

Comment:
`providedIn: 'root'` is the default value, so this can be removed

Reply:
Hrm, I don't know if that's true. https://github.com/angular/angular/blob/c461acd12e529a253b17ebfae41c4a81fb757707/packages/core/src/di/injection_token.ts#L64 Looks like it's necessary.

Reply:
line 73 sets `providedIn` to 'root' by default, right?

Reply:
Only if options is not undefined, e.g. if we passed in an empty object.

Reply:
Seems to me that it only sets "root" by default if we do pass in options. So whenever we set a default value through a factory, options are defined (options !== undefined) and it should set the `providedIn` to "root". I guess it doesn't set it to root when no options are provided, because it doesn't want to provide a token that might never need to be defined. Since the token will always need to be defined when we give it a default value (like in this case here), it makes sense to set the default to "root". It wouldn't make sense to force the app to provide a token in root if that token might not be needed.

Reply:
be4a79e3

Reply:
Oh, you're right. I missed that. I still like my example though, to make it explicitly clear.